packages/kit/src/runtime/load.js

```diff
@@ -5,9 +5,156 @@
 export function normalize(loaded) {
 	// TODO should this behaviour be dev-only?
 
-	if (loaded.error) {
-		const error = typeof loaded.error === 'string' ? new Error(loaded.error) : loaded.error;
+	if (
+		loaded.error ||
+		(loaded.status && loaded.status >= 400 && loaded.status <= 599 && !loaded.redirect)
+	) {
 		const status = loaded.status;
+		if (!loaded.error) {
+			let errorMessage = 'not assigned';
+			switch (status) {
+				case 400:
+					errorMessage = 'Bad Request';
+					break;
+				case 401:
+					errorMessage = 'Unauthorized';
+					break;
+				case 402:
+					errorMessage = 'Payment Required';
+					break;
+				case 403:
+					errorMessage = 'Forbidden';
+					break;
+				case 404:
+					errorMessage = 'Not Found';
+					break;
+				case 405:
+					errorMessage = 'Method not Allowed';
```

Comment:
should probably use a capital letter here to match the others
```suggestion
					errorMessage = 'Method Not Allowed';
```